article/app/renderers/RemoteRenderer.scala

```diff
@@ -36,19 +36,23 @@ class RemoteRenderer extends Logging {
   )(implicit request: RequestHeader): Future[Result] = {
 
     def get(): Future[Result] = {
+
       ws.url(endpoint)
         .withRequestTimeout(Configuration.rendering.timeout)
         .addHttpHeaders("Content-Type" -> "application/json")
         .post(payload)
         .map(response => {
           response.status match {
             case 200 =>
-              Cached(article)(RevalidatableResult.OkDotcomponents(Html(response.body)))
+              Cached(article)(RevalidatableResult.Ok(Html(response.body)))
+                .withHeaders("X-GU-Dotcomponents" -> "true")
             case 400 =>
-              log.error("Remote renderer validation error: " + response.body)
-              throw new Exception(response.body)
+              // if DCR returns a 400 it's because *we* failed, so frontend should return a 500
+              NoCache(play.api.mvc.Results.InternalServerError("Remote renderer validation error"))
```

Comment:
maybe you can include the fact that dotcom threw a 400 in the internal server error?